apis/kms/v1alpha1/kmskeyhandle_types.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var KMSKeyHandleGVK = GroupVersion.WithKind("KMSKeyHandle")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// KMSKeyHandleSpec defines the desired state of KMSKeyHandle
+// +kcc:proto=google.cloud.kms.v1.KeyHandle
+type KMSKeyHandleSpec struct {
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The KMSKeyHandle name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Project hosting KMSKeyHandle
+	ProjectRef *refs.ProjectRef `json:"projectRef,omitempty"`
+
+	// Location name to create KeyHandle
+	Location *string `json:"location,omitempty"`
+
+	// Indicates the resource type that the resulting [CryptoKey][] is meant to
+	// protect, e.g. `{SERVICE}.googleapis.com/{TYPE}`. See documentation for
+	// supported resource types https://cloud.google.com/kms/docs/autokey-overview#compatible-services.
+	ResourceTypeSelector *string `json:"resourceTypeSelector,omitempty"`
```

Comment:
From the controller code (no update), does it mean this field cannot be changed?

Reply:
yes